arangod/Aql/Query.cpp

```diff
@@ -1241,6 +1242,19 @@ QueryResult Query::explain() {
     Parser parser(*this, *_ast, _queryString);
     parser.parse();
 
+    // any usage of one of the following features make the query ineligible
+    // for query plan caching (at least for now):
+    if (_queryOptions.optimizePlanForCaching &&
+        (_ast->containsUpsertNode() ||
+         _ast->containsAttributeNameValueBindParameters() ||
+         _ast->containsCollectionNameValueBindParameters() ||
+         _ast->containsGraphNameValueBindParameters() ||
+         _ast->containsTraversalDepthValueBindParameters() ||
+         _ast->containsUpsertLookupValueBindParameters() ||
```

Comment:
Why isn't `containsModificationNode` checked as well? What happens if we have to `REPLACE`, `REMOVE` clauses?

Reply:
I think a "normal" modification node is fine like INSERT, REPLACE, UPDATE or REMOVE. As far as I understood, only the UPSERT is a problem. That is at least what I understood from Jan's notes.

The thing is, caching the plan is OK, if the optimization would have done exactly the same and produced the same plan if not coming from the cache. I am currently not aware that any ModificationNode would be a problem here. Are you?

Besides, I simply compied this condition from `Query::preparePlan`, as far as I see the check was just missing for `Query::explain`.